port reset_reset to Rust

benma opened the pull request.

cedwies commented on src/rust/bitbox02-rust/src/reset.rs:
```diff
@@ -0,0 +1,80 @@
+    crate::keystore::lock();
+
+    // Resetting takes longer than the default 500 ms watchdog. Bump the watchdog timeout to roughly
+    // 7 seconds (longer than needed) so we don't assume communication was lost mid-unlock.
```
"mid-unlock" or "mid-reset?"

benma replied:
Good catch. Typo as copied from keystore.rs. Reworded.

benma commented on src/rust/bitbox02-rust/src/reset.rs:
```diff
@@ -0,0 +1,80 @@
+
+    // Reset secure chip keys and U2F counter with retries, like in C reset_reset().
+    let mut reset_ok = false;
+    for _ in 0..5 {
```
Added :+1: 

This comment is AI slop :sob: I should have caught it and removed it. There is no point in mentioning C reset_reset() seeing it does not exist anymore :joy:

benma commented on src/rust/bitbox02-rust/src/hal.rs:
```diff
@@ -371,6 +383,16 @@ pub mod testing {
+
+        #[cfg(feature = "app-u2f")]
+        fn u2f_counter_set(&mut self, _counter: u32) -> Result<(), ()> {
+            Ok(())
```
Because there is no function there that reads/uses it yet.

benma commented:
@cedwies thanks for the comments. I addressed them, please check.

benma commented:
> Thanks for addressing the comments! Looks great.
> 
> Direct, focused tests of crate::reset::reset would be nice because this is a sensitive path (factory reset/auto wipe on too many failed attempts) and the control flow is new Rust glue even if underlying functions are old. However, since I see no regression versus C, I am happy to approve if we want to add tests in separate, focused PR.

Good idea, added. Please check.

benma commented on src/rust/bitbox02-rust/src/reset.rs:
```diff
@@ -0,0 +1,153 @@
+        hal.securechip.mock_reset_keys_fails();
+
+        // Simulate a non-zero U2F counter before reset.
+        SecureChip::u2f_counter_set(&mut hal.securechip, 42).unwrap();
```
Yes, in testing all features are enabled.
